lib/src/command/remove.dart

```diff
@@ -108,21 +118,28 @@ class RemoveCommand extends PubCommand {
   }
 
   Pubspec _removePackagesFromPubspec(Pubspec original, Set<String> packages) {
+    final overrides = packages
+        .where((p) => p.startsWith('override:'))
+        .map((e) => e.substring('override:'.length));
+    final otherPackages = packages.where((p) => !p.startsWith('override:'));
```

Comment:
This is argument parsing, why don't we do this in `runProtected()` and then pass the result to `_removePackagesFromPubspec` or `_writeRemovalToPubspec`

Maybe, the result can be `packagesToRemove` and `overridesToRemove` as two list of strings.